config/resource.go

```diff
@@ -136,6 +137,13 @@ func (config *Component) Validate(path string) ([]string, error) {
 	if config.Name == "" {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "name")
 	}
+	matched, err := regexp.MatchString(`^[a-zA-Z][-\w]*$`, config.Name)
```

Comment:
regex is fine but pull this out to a variable. otherwise it compiles every time

Reply:
compiled and added a comment to the pattern